notification-sdk/src/main/java/org/sunbird/notification/utils/email/SendMail.java

```diff
@@ -0,0 +1,271 @@
+      while (size > 0) {
+        message.addRecipient(Message.RecipientType.TO, new InternetAddress(emailList.get(i)));
+        i++;
+        size--;
+      }
+      message.setSubject(subject);
+      BodyPart messageBodyPart = new MimeBodyPart();
+      messageBodyPart.setContent(emailBody, "text/html; charset=utf-8");
+      // messageBodyPart.setText(mail);
+      // Create a multipar message
+      Multipart multipart = new MimeMultipart();
+      multipart.addBodyPart(messageBodyPart);
+      DataSource source = new FileDataSource(filePath);
+      messageBodyPart = null;
+      messageBodyPart = new MimeBodyPart();
+      messageBodyPart.setDataHandler(new DataHandler(source));
+      messageBodyPart.setFileName(filePath);
+      multipart.addBodyPart(messageBodyPart);
+      message.setSubject(subject);
+      message.setContent(multipart);
+      transport = session.getTransport("smtp");
+      transport.connect(host, userName, password);
+      transport.sendMessage(message, message.getAllRecipients());
+      transport.close();
+    } catch (Exception e) {
+      logger.error("Exception occured during email sending " + e, e);
+    } finally {
+      if (transport != null) {
+        try {
+          transport.close();
+        } catch (MessagingException e) {
+          logger.error("Exception occured during connection clean up.", e);
+        }
+      }
+    }
+  }
+
+  /**
+   * This method will send email with bcc.
+   *
+   * @param fromEmail fromEmail which will come in to.
+   * @param subject email subject
+   * @param body email body
+   * @param bccList recipient bcc list
+   * @return boolean
+   */
+  public boolean sendEmail(String fromEmail, String subject, String body, List<String> bccList) {
+    Transport transport = null;
+    boolean sentStatus = true;
+    try {
+      Session session = Session.getInstance(props, new GMailAuthenticator(userName, password));
+      MimeMessage message = new MimeMessage(session);
+      message.setFrom(new InternetAddress(fromEmail));
+      RecipientType recipientType = null;
+      if (bccList.size() > 1) {
+        recipientType = Message.RecipientType.BCC;
+      } else {
+        recipientType = Message.RecipientType.TO;
+      }
+      for (String email : bccList) {
+        message.addRecipient(recipientType, new InternetAddress(email));
+      }
+      if (recipientType == Message.RecipientType.BCC)
+        message.addRecipient(Message.RecipientType.TO, new InternetAddress(fromEmail));
+      message.setSubject(subject);
+      message.setContent(body, "text/html; charset=utf-8");
+      transport = session.getTransport("smtp");
+      transport.connect(host, userName, password);
+      transport.sendMessage(message, message.getAllRecipients());
+      transport.close();
+    } catch (Exception e) {
+      sentStatus = false;
+      logger.error("SendMail:sendMail: Exception occurred with message = " + e.getMessage(), e);
+    } finally {
+      if (transport != null) {
```

Comment:
I see similar code in another sendMail function. Please create a simple function or enforce a single transport for a single email sending.

Reply:
added separate method to handle Transport object.